### ISSUE TITLE
new lint: warn of dead code when range start is bigger than range end

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#4192]
````rust
fn main()  {
  (4..=1).for_each(|x| println!("{}", x));
}
````
This entire code is dead code and the content of main() is optimized away entirely without any warning.

IMO we should have a warning that suggests `(1..=4).rev()` instead

**Comment by krishna-veerareddy** [comment]
@matthiaskrgr Hey I would like to tackle this if it hasn't been assigned yet.

**Comment by matthiaskrgr** [comment]
Great, it's all yours! :)

**Comment by matthiaskrgr** [comment]
related rustc ticket: https://github.com/rust-lang/rust/issues/70925

**Comment by ebroto** [comment]
Hey @krishna-veerareddy, did you make progress on this one? I would be happy to take it if you're not working on it anymore.

**Comment by krishna-veerareddy** [comment]
@ebroto Hey I totally forgot I had this one assigned to me. I haven't started working on it so feel free to take it!
